.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -492,7 +492,16 @@ private DateTimeResolutionResult ParseComplexDatePeriod(string text, DateObject
                             datePeriodTimexType = DatePeriodTimexType.ByWeek;
                         }
 
-                        ret.Timex = TimexUtility.GenerateDatePeriodTimex(futureBegin, futureEnd, datePeriodTimexType, pastBegin, pastEnd);
+                        // If startResolution and endResolution have fuzzy years, also the combined range timex will use fuzzy years.
+                        if (startResolution.Timex.StartsWith(Constants.TimexFuzzyYear, StringComparison.Ordinal) &&
```

Comment:
I'd prefer if these two were a call to a single method that receives the necessary context and returns the appropriate timex. Can you take a stab at it and let me know if it would be too complex to change now?
If it is, please add a code comment in both place to unify later.

Reply:
I see, I can move the if statement to a new method defined in TimexUtility, that should be straightforward to do. 
Or do you mean to unify the two definitions of GenerateDatePeriodTimex in TimexUtility? There is a TODO that ask to do so but I am not sure if it is possible without breaking current behavior.

Reply:
updated method